### ISSUE TITLE
Consistency between run backend options

### ISSUE THREAD

**Issue by fntlnz** [linuxkit/linuxkit#1732]
I noted that there's some kind of incosistency between parameters in different run commands.
ATM we don't have any "plugin system" or any "common backend" or sort of thing to manage common things between linuxkit's runners, but I think that we need to address the problem now before it becomes too difficult to handle.

For example, on hyperkit 

```
  -cpus string
    	Number of CPUs (default "1")
  -diskpath string
    	Path to disk image to use
  -disksize string
    	Size of disk to create, only created if it doesn't exist
```

While on qemu
```
  -cpus int
    	Number of CPUs (default 1
  -data string
    	Metadata to pass to VM (either a path to a file or a string)
  -disk string
    	Path to disk image to used
  -disk-size int
    	Size of Disk in MB
```

Those three parameters are in both runners but:

- on qemu disk size is an integer while on hyperkit is a string (just different help message, internally all is managed) - same for cpus
- on hyperkit `diskpath` while on qemu `disk`
- on hyperkit `disksize` while on qemu `disk-size`

## Possible solutions
- some "plugin" system (if I remember correctly, @alexellis proposed something like this on the slack channel)
- Standardize the flag set (@dave-tucker suggested this in #1713)
- ....any idea?

**Comment by justincormack** [comment]
I think a plugin system may be too early - we are considering moving run to infrakit anyway, but a PR to clean up the differences (and share code where sensible) would be welcome.

**Comment by rneugeba** [comment]
This was fixed via #1758

The qemu/gcp/vmware backend don't support `data` yet, but if they do, it should be called `data` :)

Closing this for now. please reopen if there are other inconsistencies
